### DISCUSSION TITLE
Can SFTPGo access the same directories using multiple protocols?

### DISCUSSION THREAD

**Discussion by derekmahar** [drakkan/sftpgo#866] [General]
Can SFTPGo access the same directories using WebDAV, SFTP, or FTP?  In other words, could I access my directories using different types of network clients?

**Comment by drakkan** [comment]
If I understand what you mean this is the default.
You enable SFTP, FTP and WebDAV and create `user1` with home dir set to `/srv/data/sftpgo/user1`, you will see the same files with all the enabled protocols

**Reply by derekmahar** [reply]
Yes, this is what I meant.  Thank you.  That's pretty convenient.  It means I can access my files using a WebDAV client on my Android phone or tablet, or SSHFS on my Raspberry Pi 4 or Linux laptop.

**Reply by derekmahar** [reply]
By the way, have you managed to connect Windows Explorer to the WebDAV server in SFTPGo?  I tried, but it didn't work and Windows didn't report any useful error messages.

**Reply by drakkan** [reply]
> Yes, this is what I meant. Thank you. That's pretty convenient. It means I can access my files using a WebDAV client on my Android phone or tablet, or SSHFS on my Raspberry Pi 4 or Linux laptop.

Yes, this will work

**Reply by drakkan** [reply]
> By the way, have you managed to connect Windows Explorer to the WebDAV server in SFTPGo? I tried, but it didn't work and Windows didn't report any useful error messages.

It works but you need to edit some registry settings. For example basic auth is not allowed over http, only on https. Take a look [here](https://github.com/drakkan/sftpgo/blob/main/docs/webdav.md):

> if you use the native Windows client please check its usage and pay particular attention to the [registry settings](https://docs.microsoft.com/en-us/iis/publish/using-webdav/using-the-webdav-redirector#webdav-redirector-registry-settings). The default file size limit is 50MB and if you don't configure SFTPGo to use HTTPS you have to set BasicAuthLevel to 2

it is a bit late at my timezone, if you need more info I'll answer tomorrow

**Reply by derekmahar** [reply]
> it is a bit late at my timezone, if you need more info I'll answer tomorrow

No worries or hurry.  Have a good night!

**Reply by derekmahar** [reply]
I managed to map a Windows drive to my WebDAV account without changing any registry settings.  This time I entered the HTTPS URL of my WebDAV server into the drive mapping dialog instead of following the instructions at "Connect to a Web site that you can use to store your documents and pictures."  However, the local network URL didn't work presumably because the Windows WebDAV client fails to connect when it encounters a certificate that doesn't match the domain name in the URL.  Is it possible to configure the Windows WebDAV client to ignore certificate errors?

**Reply by derekmahar** [reply]
The instructions in https://windowsloop.com/map-webdav-drive-in-windows-10/ didn't work for me, though.

**Reply by drakkan** [reply]
> I managed to map a Windows drive to my WebDAV account without changing any registry settings. This time I entered the HTTPS URL of my WebDAV server into the drive mapping dialog instead of following the instructions at "Connect to a Web site that you can use to store your documents and pictures." However, the local network URL didn't work presumably because the Windows WebDAV client fails to connect when it encounters a certificate that doesn't match the domain name in the URL. 

try adding an entry in your hosts file. This way you can have a match between the certificate name and the domain name in the URL

> Is it possible to configure the Windows WebDAV client to ignore certificate errors?

I'm not a windows expert, a solution is to import the root ca in the trusted ones. I don't know if other workarounds are available

**Reply by drakkan** [reply]
> The instructions in https://windowsloop.com/map-webdav-drive-in-windows-10/ didn't work for me, though.

I just tested these instructions and they work for me. I'm connecting to SFTPGo over https with a valid certificate

**Reply by derekmahar** [reply]
> > The instructions in https://windowsloop.com/map-webdav-drive-in-windows-10/ didn't work for me, though.
> I just tested these instructions and they work for me. I'm connecting to SFTPGo over https with a valid certificate

Are you running Windows 10 or 11?  My domain also has a valid certificate, but Windows seems to believe the certificate is invalid or doesn't match the domain name.  Not the end of the world.  I can still map a drive to my WebDAV endpoint using the regular Windows Explorer drive mapping dialog or from the command line using `NET USE`.

**Reply by drakkan** [reply]
> > > The instructions in https://windowsloop.com/map-webdav-drive-in-windows-10/ didn't work for me, though.
> > > I just tested these instructions and they work for me. I'm connecting to SFTPGo over https with a valid certificate
> 
> Are you running Windows 10 or 11? My domain also has a valid certificate, but Windows seems to believe the certificate is invalid or doesn't match the domain name. Not the end of the world. I can still map a drive to my WebDAV endpoint using the regular Windows Explorer drive mapping dialog or from the command line using `NET USE`.

Windows 10. I have no Windows 11 VM for now
